components/metrics/http_test.go

```diff
@@ -0,0 +1,65 @@
+package metrics_test
+
+import (
+	"net/http"
+	"testing"
+	"time"
+
+	"github.com/ThreeDotsLabs/watermill/components/metrics"
+	"github.com/pkg/errors"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestCreateRegistryAndServeHTTP_metrics_endpoint(t *testing.T) {
+	reg, cancel := metrics.CreateRegistryAndServeHTTP(":8090")
+	defer cancel()
+	err := reg.Register(prometheus.NewBuildInfoCollector())
+	if err != nil {
+		t.Fatal(errors.Wrap(err, "registration of prometheus build info collector failed"))
+	}
+	waitServerReady(t, "http://localhost:8090")
+	resp, err := http.DefaultClient.Get("http://localhost:8090/metrics")
+	if resp != nil {
```

Comment:
+1 for checking the resp before the error :)

Reply:
In fact I did this by habit, however if both err and response are returned, you don't have to close the body.
https://stackoverflow.com/questions/32818472/do-we-need-to-close-the-response-object-if-an-error-occurs-while-calling-http-ge